numba/tests/test_override.py

```diff
@@ -0,0 +1,58 @@
+import unittest
+
+from numba.tests.support import (TestCase, override_config)
+from numba import njit
+from numba.core import types
+import llvmlite.binding as llvm
+
+
+class TestPassManagerOptimization(TestCase):
+    """ Tests that pass manager is not overriding the intended
+    optimization level.
+    """
+
+    def _get_llvmir(self, fn, sig):
+        with override_config('OPT', 0):
+            fn.compile(sig)
+            return fn.inspect_llvm(sig)
+
+    def test_override_config(self):
+        @njit(debug=True, error_model='numpy')
+        def foo(a):
+            b = a + 1.23
+            c = b * 2.34
+            d = b / c
+            print(d)
+            return d
+
+        sig = (types.float64,)
+        full_ir = self._get_llvmir(foo, sig=sig)
+
+        module = llvm.parse_assembly(full_ir)
+
+        name = foo.overloads[foo.signatures[0]].fndesc.mangled_name
+        funcs = [x for x in module.functions if x.name == name]
+        self.assertEqual(len(funcs), 1)
+        func = funcs[0]
+        blocks = [x for x in func.blocks]
+        self.assertGreater(len(blocks), 1)
+        block = blocks[0]
+
+        # Find sequence without non-debug instructions
```

Comment:
```suggestion
        # Find sequence with non-debug instructions
```
I think this is looking for the point at which the arithmetic instructions start and is discounting `call` as those are associated with `llvm.dbg.declare`.